File: output/GPT4o-Mini/xCodeEval_4000/C++/C/2af7e700e817e767244391f8ba5f3862.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>
#include <string.h>
#include <assert.h>
#include <limits.h>
#include <stdbool.h>

#define MAX_N 100010
#define EPS 1e-8
#define REP(a,b) for(long long (a) = 0; (a) < (b); ++(a))

typedef struct {
    int hoge[5];
} Vert;

bool sign(Vert a, Vert b) {
    int hoge = 0;
    REP(q, 5) {
        hoge += a.hoge[q] * b.hoge[q];
    }
    return hoge <= 0;
}

void solve() {
    int n;
    scanf("%d", &n);
    Vert inputs[MAX_N];
    REP(i, n) {
        REP(q, 5) {
            scanf("%d", &inputs[i].hoge[q]);
        }
    }
    int ans[MAX_N];
    int ans_size = 0;
    REP(i, n) {
        int ok = 1;
        REP(q, n) {
            if (i == q) continue;
            for (int j = q + 1; j < n; ++j) {
                if (i == j) continue;
                Vert A, B;
                REP(t, 5) {
                    A.hoge[t] = inputs[q].hoge[t] - inputs[i].hoge[t];
                    B.hoge[t] = inputs[j].hoge[t] - inputs[i].hoge[t];
                }
                if (sign(A, B) == 0) {
                    ok = 0;
                    break;
                }
            }
            if (ok == 0) break;
        }
        if (ok == 1) {
            ans[ans_size++] = i;
        }
    }
    printf("%d\n", ans_size);
    REP(i, ans_size) {
        printf("%d ", ans[i] + 1);
    }
    printf("\n");
}

int main() {
    solve();
    return 0;
}

File: output/GPT4o-Mini/xCodeEval_4000/C++/Java/2af7e700e817e767244391f8ba5f3862.java
import java.util.*;
import java.io.*;

public class 2af7e700e817e767244391f8ba5f3862_nc {
    static class Vert {
        int[] hoge = new int[5];
    }

    static boolean sign(Vert a, Vert b) {
        int hoge = 0;
        for (int q = 0; q < 5; q++) {
            hoge += a.hoge[q] * b.hoge[q];
        }
        return hoge <= 0;
    }

    static void solve() {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        List<Vert> inputs = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            Vert a = new Vert();
            for (int q = 0; q < 5; q++) {
                a.hoge[q] = scanner.nextInt();
            }
            inputs.add(a);
        }
        List<Integer> ans = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            boolean ok = true;
            for (int q = 0; q < n; q++) {
                if (i == q) continue;
                for (int j = q + 1; j < n; j++) {
                    if (i == j) continue;
                    Vert A = new Vert(), B = new Vert();
                    for (int t = 0; t < 5; t++) {
                        A.hoge[t] = inputs.get(q).hoge[t] - inputs.get(i).hoge[t];
                        B.hoge[t] = inputs.get(j).hoge[t] - inputs.get(i).hoge[t];
                    }
                    if (!sign(A, B)) {
                        ok = false;
                        break;
                    }
                }
                if (!ok) break;
            }
            if (ok) {
                ans.add(i);
            }
        }
        System.out.println(ans.size());
        for (int i = 0; i < ans.size(); i++) {
            System.out.print((ans.get(i) + 1) + " ");
        }
        System.out.println();
    }

    public static void main(String[] args) {
        solve();
    }
}
